src/format.rs

```diff
@@ -77,7 +77,7 @@ impl fmt::Display for HumanDuration {
         for (i, &(cur, _, _)) in UNITS.iter().enumerate() {
             idx = i;
             match UNITS.get(i + 1) {
-                Some(&next) if self.0 + next.0 / 2 >= cur + cur / 2 => break,
+                Some(&next) if self.0.saturating_add(next.0 / 2) >= cur + cur / 2 => break,
```

Comment:
Do we also need `saturating_add()` for `cur` here?

Reply:
I don't think so, `cur` comes from a list of constants and none of them make it overflow.